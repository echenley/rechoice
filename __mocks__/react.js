const React = require.requireActual('react')

React.Fragment = 'Fragment'

module.exports = React
