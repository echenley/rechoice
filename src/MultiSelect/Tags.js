// @flow

import * as React from 'react'
import type { Item, TagRenderer } from '../types'

type Props = {
  className: string,
  itemKey: string,
  removeTag: (item: Item) => mixed,
  renderTag: TagRenderer,
  value: Array<Item>,
}

class Tags extends React.Component<Props> {
  renderTag = (item: Item) => {
    const { itemKey, removeTag, renderTag } = this.props

    const tag = renderTag({
      item,
      removeTag: () => removeTag(item),
    })

    return React.cloneElement(tag, {
      key: item[itemKey],
    })
  }

  render() {
    const { className, value } = this.props
    return <span className={className}>{value.map(this.renderTag)}</span>
  }
}

export default Tags
