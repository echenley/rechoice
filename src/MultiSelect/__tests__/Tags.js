import * as React from 'react'
import { shallow } from 'enzyme'
import Tags from '../Tags'

const renderTag = ({ item, removeTag }) => (
  <button onClick={removeTag}>{`Item ${item.id}`}</button>
)

const renderComponent = props => {
  return shallow(
    <Tags
      className="tags"
      itemKey="id"
      removeTag={() => {}}
      renderTag={renderTag}
      value={[{ id: 1 }, { id: 2 }]}
      {...props}
    />
  )
}

describe('Tags', () => {
  it('renders correctly', () => {
    expect(renderComponent().getElement()).toMatchSnapshot()
  })

  it('calls onChange with updated value when removeTag is called', () => {
    const removeTag = jest.fn()
    const component = renderComponent({ removeTag })

    component
      .find('button')
      .first()
      .simulate('click')

    expect(removeTag).toHaveBeenCalledWith({ id: 1 })
  })
})
