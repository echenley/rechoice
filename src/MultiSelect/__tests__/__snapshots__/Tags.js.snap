// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
<span
  className="tags"
>
  <button
    onClick={[Function]}
  >
    Item 1
  </button>
  <button
    onClick={[Function]}
  >
    Item 2
  </button>
</span>
`;
